src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/IdentifiersShouldHaveCorrectSuffix.cs

```diff
@@ -176,14 +176,15 @@ private static void AnalyzeCompilationStart(CompilationStartAnalysisContext cont
             foreach (var type in typeSymbols)
             {
                 // User specific mapping has higher priority than hardcoded one
-                if (userMap.TryGetValue(type.OriginalDefinition, out var suffix) &&
-                    !RoslynString.IsNullOrWhiteSpace(suffix))
+                if (userMap.TryGetValue(type.OriginalDefinition, out var suffix))
                 {
-                    suffixInfo = SuffixInfo.Create(suffix, canSuffixBeCollection: false);
-                    return true;
+                    if (!RoslynString.IsNullOrWhiteSpace(suffix))
```

Comment:
So we are still going through all symbols even after we found an empty match in userMap? The behavior makes sense, just consider adding a test to capture this intention. (i.e. some type implement two interfaces, one with user overriding the other without)